src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -946,12 +948,22 @@ void MetaspaceShared::initialize_runtime_shared_and_meta_spaces() {
     _requested_base_address = static_mapinfo->requested_base_address();
     if (dynamic_mapped) {
       FileMapInfo::set_shared_path_table(dynamic_mapinfo);
+      // turn AutoCreateSharedArchive off if successfully mapped
+      if (AutoCreateSharedArchive) {
+        AutoCreateSharedArchive = false;
+      }
     } else {
       FileMapInfo::set_shared_path_table(static_mapinfo);
     }
   } else {
     set_shared_metaspace_range(NULL, NULL, NULL);
     UseSharedSpaces = false;
+    if (AutoCreateSharedArchive) {
+      AutoCreateSharedArchive = false;
+    }
+    if (DynamicDumpSharedSpaces) {
+      DynamicDumpSharedSpaces = false;
+    }
```

Comment:
I think there's no need to put the above checks in 'if' blocks. I would suggest replacing the above six lines with:
```
// The base archive cannot be mapped. We cannot dump the dynamic shared archive.
AutoCreateSharedArchive = false;
DynamicDumpSharedSpaces = false;
```